"Check Northbound, Storm and Topology Engine logging" test fails

Reproduced on Jenkins job which runs functional tests.
I could not reproduce it locally.

Test fails with following logs:

```
2019-01-31 14:33:10.935 [INFO] Running org.openkilda.functionaltests.spec.logging.CheckLoggingSpec
2019-01-31 14:33:10.936 2019-01-31T10:33:10,893 INFO  TestDataLoggingExtension:-1 - Running test: Check Floodlight logging
2019-01-31 14:33:11.132 2019-01-31T10:33:11,130 INFO  TestDataLoggingExtension:-1 - Running test: Check Northbound, Storm and Topology Engine logging
2019-01-31 14:36:14.639 2019-01-31T10:36:14,642 INFO  TestDataLoggingExtension:-1 - Running spec: ThrottlingRerouteSpec
2019-01-31 14:36:14.640 [ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 183.76 s <<< FAILURE! - in org.openkilda.functionaltests.spec.logging.CheckLoggingSpec
2019-01-31 14:36:14.643 [ERROR] Check Northbound, Storm and Topology Engine logging(org.openkilda.functionaltests.spec.logging.CheckLoggingSpec)  Time elapsed: 183.459 s  <<< FAILURE!
2019-01-31 14:36:14.644 org.spockframework.runtime.ConditionFailedWithExceptionError: 
2019-01-31 14:36:14.645 Condition failed with Exception:
2019-01-31 14:36:14.645 
2019-01-31 14:36:14.646 Wrappers.wait(timeout, 10) { def nbLogs = elastic.getLogs(new ElasticQueryBuilder().setTags(KildaTags.NORTHBOUND). setTimeRange(timeout * 2).setLevel("WARN").build()) def stormLogs = elastic.getLogs(new ElasticQueryBuilder().setTags(KildaTags.STORM_WORKER). setTimeRange(timeout * 2).setLevel("WARN").build()) def tpLogs = elastic.getLogs(new ElasticQueryBuilder().setTags(KildaTags.TOPOLOGY_ENGINE). setTimeRange(timeout * 2).setLevel("INFO").build()) assert nbLogs?.hits?.hits?.any { hit -> hit.source.message.contains(flowId) }: "Northbound should generate an error message about not being able to find a flow" assert stormLogs?.hits?.hits?.any { hit -> hit.source.message.contains(flowId) }: "Storm should generate an error message about not being able to find a flow" assert tpLogs?.hits?.hits?.any { hit -> hit.source.message.contains(switchId.toString()) }: "Topology Engine should generate an info message in case of a switch rules validation event" }
2019-01-31 14:36:14.648 
2019-01-31 14:36:14.648 	at org.openkilda.functionaltests.spec.logging.CheckLoggingSpec.Check Northbound, Storm and Topology Engine logging(CheckLoggingSpec.groovy:46)
2019-01-31 14:36:14.649 Caused by: org.openkilda.functionaltests.helpers.Wrappers$WaitTimeoutException: 
2019-01-31 14:36:14.649 Condition was not satisfied within 180.0 seconds
2019-01-31 14:36:14.650 Failed with exception:
2019-01-31 14:36:14.651 
2019-01-31 14:36:14.651 Condition not satisfied:
2019-01-31 14:36:14.652 
2019-01-31 14:36:14.652 stormLogs?.hits?.hits?.any { hit -> hit.source.message.contains(flowId) }
2019-01-31 14:36:14.653 
2019-01-31 14:36:14.653 Storm should generate an error message about not being able to find a flow
2019-01-31 14:36:14.654 
2019-01-31 14:36:14.654 	at org.openkilda.functionaltests.spec.logging.CheckLoggingSpec.Check Northbound, Storm and Topology Engine logging(CheckLoggingSpec.groovy:46)
2019-01-31 14:36:14.655 
```

--- a/services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/logging/CheckLoggingSpec.groovy
+++ b/services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/logging/CheckLoggingSpec.groovy
@@ -2,6 +2,8 @@ package org.openkilda.functionaltests.spec.logging
 
 import org.openkilda.functionaltests.BaseSpecification
 import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.error.MessageError
+import org.openkilda.model.SwitchId
 import org.openkilda.testing.service.elastic.ElasticQueryBuilder
 import org.openkilda.testing.service.elastic.ElasticService
 import org.openkilda.testing.service.elastic.model.KildaTags
@@ -9,8 +11,6 @@ import org.openkilda.testing.service.elastic.model.KildaTags
 import groovy.util.logging.Slf4j
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.web.client.HttpClientErrorException
-import spock.lang.Ignore
-import spock.lang.Issue
 import spock.lang.Narrative
 
 @Slf4j
@@ -33,37 +33,41 @@ class CheckLoggingSpec extends BaseSpecification {
         result.hits.hits.any { hit -> hit.source.message.toLowerCase().contains(discoveryMessage) }
     }
 
-    @Issue("https://github.com/telstra/open-kilda/issues/1966")
-    @Ignore
-    def "Check Northbound, Storm and Topology Engine logging"() {
+    def "Check Northbound and Storm logging"() {
+        when: "A non-existent switch is requested"
+        def switchId = new SwitchId("123456789")
+        northbound.getSwitch(switchId)
+
+        then: "An error is received (404 code)"
+        def switchExc = thrown(HttpClientErrorException)
+        def switchErrorMsg = "Switch $switchId not found"
+
+        switchExc.rawStatusCode == 404
+        switchExc.responseBodyAsString.to(MessageError).errorMessage.contains(switchErrorMsg)
+
         when: "A non-existent flow is requested"
-        int timeout = 180
-        def flowId = "nonexistentflowid" + System.currentTimeMillis()
-        try {
-            northbound.getFlow(flowId)
-        } catch (HttpClientErrorException e) {
+        def flowId = "nonexistentFlowId" + System.currentTimeMillis()
+        northbound.getFlow(flowId)
 
-        }
+        then: "An error is received (404 code)"
+        def flowExc = thrown(HttpClientErrorException)
+        def flowErrorMsg = "Can not get flow: Flow $flowId not found"
 
-        and: "Rules on a switch are validated"
-        def switchId = topology.activeSwitches.first().dpId
-        northbound.validateSwitchRules(switchId)
+        flowExc.rawStatusCode == 404
+        flowExc.responseBodyAsString.to(MessageError).errorMessage.contains(flowErrorMsg)
 
-        then: "Northbound, Storm and Topology Engine should log these actions within 3 minutes"
+        and: "Northbound and Storm should log these actions within 30 seconds"
+        int timeout = 31
         Wrappers.wait(timeout, 10) {
             def nbLogs = elastic.getLogs(new ElasticQueryBuilder().setTags(KildaTags.NORTHBOUND).
-                    setTimeRange(timeout * 2).setLevel("WARN").build())
+                    setTimeRange(timeout * 2).setLevel("ERROR").build())
             def stormLogs = elastic.getLogs(new ElasticQueryBuilder().setTags(KildaTags.STORM_WORKER).
                     setTimeRange(timeout * 2).setLevel("WARN").build())
-            def tpLogs = elastic.getLogs(new ElasticQueryBuilder().setTags(KildaTags.TOPOLOGY_ENGINE).
-                    setTimeRange(timeout * 2).setLevel("INFO").build())
 
-            assert nbLogs?.hits?.hits?.any { hit -> hit.source.message.contains(flowId) }:
-                    "Northbound should generate an error message about not being able to find a flow"
-            assert stormLogs?.hits?.hits?.any { hit -> hit.source.message.contains(flowId) }:
-                    "Storm should generate an error message about not being able to find a flow"
-            assert tpLogs?.hits?.hits?.any { hit -> hit.source.message.contains(switchId.toString()) }:
-                    "Topology Engine should generate an info message in case of a switch rules validation event"
+            assert nbLogs?.hits?.hits?.any { hit -> hit.source.message.contains(switchErrorMsg) }:
+                    "Northbound should generate an error message about not being able to find the switch"
+            assert stormLogs?.hits?.hits?.any { hit -> hit.source.message.contains(flowErrorMsg) }:
+                    "Storm should generate an error message about not being able to find the flow"
         }
     }
 }

--- a/services/src/test-library/src/main/java/org/openkilda/testing/service/northbound/NorthboundServiceImpl.java
+++ b/services/src/test-library/src/main/java/org/openkilda/testing/service/northbound/NorthboundServiceImpl.java
@@ -86,16 +86,8 @@ public class NorthboundServiceImpl implements NorthboundService {
 
     @Override
     public FlowPayload getFlow(String flowId) {
-        try {
-            return restTemplate.exchange("/api/v1/flows/{flow_id}", HttpMethod.GET,
-                    new HttpEntity(buildHeadersWithCorrelationId()), FlowPayload.class, flowId).getBody();
-        } catch (HttpClientErrorException ex) {
-            if (ex.getStatusCode() != HttpStatus.NOT_FOUND) {
-                throw ex;
-            }
-
-            return null;
-        }
+        return restTemplate.exchange("/api/v1/flows/{flow_id}", HttpMethod.GET,
+                new HttpEntity(buildHeadersWithCorrelationId()), FlowPayload.class, flowId).getBody();
     }
 
     @Override

